internal/core/services/rooms/room_manager.go

```diff
@@ -426,20 +426,37 @@ func (m *RoomManager) createRoomOnStorageAndRuntime(ctx context.Context, schedul
 }
 
 func (m *RoomManager) populateSpecWithHostPort(scheduler entities.Scheduler) (*game_room.Spec, error) {
-	numberOfPorts := 0
 	spec := scheduler.Spec.DeepCopy()
-	for _, container := range spec.Containers {
-		numberOfPorts += len(container.Ports)
-	}
-	allocatedPorts, err := m.PortAllocator.Allocate(scheduler.PortRange, numberOfPorts)
-	if err != nil {
-		return nil, err
-	}
-	portIndex := 0
-	for _, container := range spec.Containers {
-		for i := range container.Ports {
-			container.Ports[i].HostPort = int(allocatedPorts[portIndex])
-			portIndex++
+
+	// Backwards compatibility for legacy port range configuration
+	if scheduler.PortRange != nil {
```

Comment:
nit: what if we ported every spec.PortRange config to the container port range new property and use that to calculate all allocations instead of doing conditionally by each property? would that break game that do not have the container.port new config?